pageserver/src/layered_repository.rs

```diff
@@ -102,6 +103,22 @@ lazy_static! {
     .expect("failed to define a metric");
 }
 
+/// The name of the metadata file pageserver creates per timeline.
+pub const METADATA_FILE_NAME: &str = "metadata";
```

Comment:
Since I start to deal with metadata more and more, time to extract it into a separate constant at least.
Ideally, I think we should extract this + `filename.rs` contents + various layer name `path()` methods into some struct that manages all layered storage workdir access.
But that's a separate big task.

Reply:
Maybe introduce this constant and the metadata_path function in a separate PR or commit, for easier review, even if you don't do any further refactoring right now?